send-transaction-service/src/send_transaction_service.rs

```diff
@@ -469,17 +506,19 @@ impl SendTransactionService {
 
         if !batched_transactions.is_empty() {
             // Processing the transactions in batch
-            let addresses = Self::get_tpu_addresses(tpu_address, leader_info, config);
-
             let wire_transactions = transactions
                 .iter()
                 .filter(|(signature, _)| batched_transactions.contains(signature))
                 .map(|(_, transaction_info)| transaction_info.wire_transaction.as_ref())
                 .collect::<Vec<&[u8]>>();
 
-            for address in &addresses {
-                let iter = wire_transactions.chunks(config.batch_size);
-                for chunk in iter {
+            let iter = wire_transactions.chunks(config.batch_size);
+            for chunk in iter {
+                let mut leader_info_provider = leader_info_provider.lock().unwrap();
+                let leader_info = leader_info_provider.get_leader_info();
+                let addresses = Self::get_tpu_addresses(tpu_address, leader_info, config);
+
+                for address in &addresses {
                     Self::send_transactions(address, chunk);
                 }
```

Comment:
What do you think about dropping `leader_info_provider` explicitly so that this doesn't hold the lock the entire block? ie:
```suggestion
                let mut leader_info_provider = leader_info_provider.lock().unwrap();
                let leader_info = leader_info_provider.get_leader_info();
                drop(leader_info_provider);
                let addresses = Self::get_tpu_addresses(tpu_address, leader_info, config);

                for address in &addresses {
                    Self::send_transactions(address, chunk);
                }
```
Obviously, we hope that send_transactions doesn't take very long, but might as well block `send_transactions_in_batch()` on L295 as little as possible

Reply:
need to at least hold it through the address query, right?

i'd prefer a scope over manual drop for improved defensiveness

Reply:
Oops, yes, sorry, through the address generation due to the ref.
Sure, scope is great!